pkg/validation/validate.go

```diff
@@ -135,11 +149,23 @@ func ValidateLabels(limits LabelValidationLimits, tenantID string, ls []*typesv1
 		if len(l.Value) > limits.MaxLabelValueLength(tenantID) {
 			return NewErrorf(LabelValueTooLong, LabelValueTooLongErrorMsg, phlaremodel.LabelPairsString(ls), l.Value)
 		}
-		var origName string
+		var origName, sanitized string
 		var ok bool
-		if origName, l.Name, ok = SanitizeLabelName(l.Name); !ok {
+		if origName, sanitized, ok = SanitizeLabelName(l.Name); !ok {
 			return NewErrorf(InvalidLabels, InvalidLabelsErrorMsg, phlaremodel.LabelPairsString(ls), "invalid label name '"+origName+"'")
 		}
+
+		// Check if a dup would be created because of sanitization
+		if origName != sanitized {
+			if sanitizedNames[sanitized] {
+				// Add "_dup" suffix to the conflicting sanitized label name
+				sanitized = sanitized + "_dup"
+			} else {
+				sanitizedNames[sanitized] = true
+			}
+		}
+		l.Name = sanitized
```

Comment:
Changing the name requires another `sort.Sort(phlaremodel.Labels(ls))` to ensure the order: 

Let's say I have I have label names with `service_name_aaaa` and `service_name_eee` , `service_name_dup` should be in-between them.

Reply:
Yes, I somewhat miss to reorder the labels after adding the sanitised name, good catch.